### PR TITLE
Leverage PROJECT_CWD to set cwd in yarn 3 bin "proxies"

### PR DESCRIPTION
**Pull request by jtoar**
To upgrade RW projects to yarn 3, we had to make bin "proxy" files so that we could still access the bins of our dependencies. See https://github.com/redwoodjs/redwood/pull/4444 for the full break down.

When you run a bin, yarn 3 sets the cwd to the workspace calling it. This revealed a hidden dependency in many of our CLI commands that expect the cwd to be the root, so we imported `getPaths` from `@redwoodjs/cli` (an export of the one in `@redwoodjs/internal`) in many of the bin proxy files to find the root and change to it.

This works and has worked for a long time, but turns out to be a not-so-great idea because it means pulling in a massive amount of `@redwoodjs/internal` for any CLI command, and possibly introducing some indeterminacy around the way env vars are loaded.

Instead of importing `getPaths`, this PR leverages the `PROJECT_CWD` env var provided by yarn 3. It's value is what you expect. See https://yarnpkg.com/advanced/lifecycle-scripts/#environment-variables.

Next step is to confirm whether or not this change fixes https://github.com/redwoodjs/redwood/issues/5657.

Here's a visualization of the change using `0x` to generate a flamegraph:

bin-proxies

The exact command for reproduction is

```
yarn dlx 0x -o ~/your-rw-project/node_modules/@redwoodjs/core/dist/bins/redwood.js --help
```

### CODE REVIEW

**Comment by dac09** [comment]
Yarn master Dom at it again, nice one!

File: packages/cli/src/index.js [hunk 1]
```diff
@@ -79,7 +79,7 @@ const loadDotEnvDefaultsMiddleware = () => {
   config({
     path: path.join(getPaths().base, '.env'),
     defaults: path.join(getPaths().base, '.env.defaults'),
-    encoding: 'utf8',
```
Comment (jtoar) [inline packages/cli/src/index.js:82]: This is the default value.

File: packages/cli/src/index.js [hunk 2]
```diff
@@ -79,7 +79,7 @@ const loadDotEnvDefaultsMiddleware = () => {
...
+    multiline: true,
```
Comment (jtoar) [inline packages/cli/src/index.js:82]: Partly fixes https://github.com/redwoodjs/redwood/issues/5657

File: packages/core/src/bins/redwood.ts [hunk 3]
```diff
@@ -1,37 +1,33 @@
...
 const requireFromCli = createRequire(
   require.resolve('@redwoodjs/cli/package.json')
 )
 
 const bins = requireFromCli('./package.json')['bin']
 
-// Ensure we run all commands from the base of the RW project
-// even if you invoke from ./web or ./api
-const rwProjectRoot = requireFromCli('./dist/lib/index.js').getPaths().base
-process.chdir(rwProjectRoot)
```
Comment (jtoar) [inline packages/core/src/bins/redwood.ts:35]: This is already at the root, so no need to change to it

**Comment by jtoar** [comment]
I've tested this locally and it works so far, so I'm going to merge once checks past and test again in canary.

File: packages/cli/src/index.js (hunk 2 above)
Comment (dac09) [reply packages/cli/src/index.js:82]: Do you think we could add a test for this?

Comment (jtoar) [reply packages/cli/src/index.js:82]: Sorry I didn't see this comment; I'll revisit

Comment (dac09) [reply packages/cli/src/index.js:82]: No worries! 👍👍👍

**Pull request merged**